File: src/main/java/io/github/dqualizer/dqlang/types_new/adapter/options/Scenario.java
package io.github.dqualizer.dqlang.types_new.adapter.options;

import com.fasterxml.jackson.annotation.JsonSubTypes;
import com.fasterxml.jackson.annotation.JsonTypeInfo;

@JsonTypeInfo(
        use = JsonTypeInfo.Id.NAME,
        property = "executor")
@JsonSubTypes({
        @JsonSubTypes.Type(value = ConstantScenario.class, name = "constant-vus"),
        @JsonSubTypes.Type(value = RampingScenario.class, name = "ramping-vus")
})
public abstract class Scenario {
}

File: src/main/kotlin/io/github/dqualizer/dqlang/types/adapter/k6/options/Scenario.kt
package io.github.dqualizer.dqlang.types.adapter.k6.options

import com.fasterxml.jackson.annotation.JsonSubTypes
import com.fasterxml.jackson.annotation.JsonTypeInfo

@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "executor")
@JsonSubTypes(
    JsonSubTypes.Type(value = ConstantScenario::class, name = "constant-vus"),
    JsonSubTypes.Type(value = RampingScenario::class, name = "ramping-vus")
)
sealed class Scenario
